storage/aptosdb/src/state_kv_db.rs

```diff
@@ -137,6 +141,30 @@ impl StateKvDb {
         )
     }
 
+    pub(crate) fn create_checkpoint(
+        db_root_path: impl AsRef<Path>,
+        cp_root_path: impl AsRef<Path>,
+    ) -> Result<()> {
+        let state_kv_db = Self::open(db_root_path, RocksdbConfig::default(), false)?;
+        let cp_state_kv_db_path = cp_root_path.as_ref().join(STATE_KV_DB_FOLDER_NAME);
+        std::fs::remove_dir_all(&cp_state_kv_db_path).unwrap_or(());
+        std::fs::create_dir_all(&cp_state_kv_db_path).unwrap_or(());
+        let sharding = false;
+        if sharding {
+            for shard_id in 0..NUM_STATE_SHARDS {
+                state_kv_db
+                    .db_shard(shard_id as u8)
+                    .create_checkpoint(Self::db_shard_path(
+                        cp_root_path.as_ref(),
+                        shard_id as u8,
+                    ))?;
+            }
+        }
+        state_kv_db
+            .metadata_db()
+            .create_checkpoint(Self::metadata_db_path(cp_root_path))
```

Comment:
My intuition: you need to checkpoint the metadata first? (Assuming one day we'll be able to checkpoint on the side while the node is running)

Reply:
Done